model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -77,57 +77,57 @@ async def generate(request: Request) -> Response:
     request_id = random_uuid()
     try:
         results_generator = engine.generate(prompt, sampling_params, request_id)
-    except AsyncEngineDeadError as e:
-        print(f"The vllm engine is dead, exiting the pod: {e}")
-        exit(1)
 
-    # Streaming case
-    async def stream_results() -> AsyncGenerator[str, None]:
+        # Streaming case
+        async def stream_results() -> AsyncGenerator[str, None]:
```

Comment:
now i think the type annotation should be AsyncIterator?

Reply:
I think AsyncGenerator is still correct?